src/historywork/BatchWork.h

```diff
@@ -17,6 +18,13 @@ class BatchWork : public Work
        they'd like to perform. This class only acts as a commander, adding more
        work if it has bandwidth.
     **/
+    void handleNewWork(bool isFirst);
+
+    // Keep track of last assigned work as any next work will depend on it
+    // (has to be strong reference in case it finished prior to new work
+    // assigned)
+    std::shared_ptr<BatchableWork> mLastAssignedWork;
```

Comment:
I don't think this is the right change:
it seems to me that it would be much cleaner to have `yieldMoreWork` manage dependencies between work items as it's the iterator, and it knows what it's producing (so if we use lambdas for notifications, we have access to the right types).

Right now having `BatchWork` doing it hard codes for no good reason the dependency between work items.

Reply:
This probably requires some changes to `BatchableWork` to enable those semantics, but we would end up with much cleaner contract and semantics:
* allow more complex synchronization by allowing any `BatchableWork` to tell its associated `BatchWork` that it's waiting on an existing `BatchableWork` after the fact (again, by having dependencies tracked at the generator level, it's easy to keep track of the right `BatchableWork`)